src/mca/pcompress/zlib/compress_zlib.c

```diff
@@ -167,7 +169,7 @@ static bool doit(uint8_t **outbytes, size_t len2, const uint8_t *inbytes, size_t
 
     rc = inflate(&strm, Z_FINISH);
     inflateEnd(&strm);
-    if (Z_OK == rc) {
+    if (Z_STREAM_END == rc) {
```

Comment:
Sorry to be a nag, but could you again point to the documentation where this doesn't return `Z_OK`? I'm beginning to wonder if this is perhaps something reflecting a difference in zlib version and might require some configuration checks to support. I wrote the `zlib` plugin quite some time ago and it is entirely possible things have changed.

Reply:
Sure, here's the link: https://github.com/madler/zlib/blob/master/zlib.h#L462-L463

Reply:
Is it possible this has changed at some point from prior versions? I see the master doc supports the proposed change - just want to ensure that some earlier version didn't return a different value so we don't get into trouble. We can detect versions during configure and protect if necessary.

Reply:
And also this one (https://github.com/madler/zlib/blob/master/zlib.h#L502-L504), which mentions that Z_OK is not exactly an error but that more input bytes are expected. In our case, all input bytes should be available. Z_OK would actually mean that the message has only partially arrived or was truncated for another reason.

Reply:
I have blamed zlib.h back to 0.79, which indicates that the API did not change: https://github.com/madler/zlib/blame/913afb9174bb474104049906c1382dec81826424/zlib.h

Also I did not see version checks in other projects which I have seen so far in this regard, e.g. https://sources.debian.org/src/firefox/101.0.1-1/dom/fetch/FetchUtil.cpp/?hl=634#L634 (I can definitely recommend https://codesearch.debian.net to quickly scan through open source projects).

Reply:
Okay - thanks for checking for me! I guess I simply misunderstood whatever it was I read. Appreciate the correction!